Fixed PrintRopeStats

canelhasmateus opened the pull request:
PrintRopeCacheStats is declared as a const false,
that acted as a guardian to a when block that references undeclared variables.
As such, setting it to true crashes the compiler.
Importing ropes.nim fixes the issue.

saem commented:
Hi @canelhasmateus -- thanks for the PR (sorry about the template 😞)

Quick thing for the commit message, could you format it according to the contribution guidelines, [direct link to relevant section](https://nim-works.github.io/nimskull/contributing.html#pr-and-commit-message-guidelines-commit-message-guidelines). Relevant bit being first line is 50 chars or less, and body is 72 chars or less per line.

----
Aside, sorry you're running into the first use of the PR template. 😬 Bors will shove that into the merge PR, it's probably easy enough to fix up the commit message first, then copy that into the PR body -- mind the line/thematic break for CI.

Obviously the text of the PR template isn't clear enough, anything that might help you arrive at the above conclusions yourself? I need some of your feedback with it to help fix. Thanks.

canelhasmateus commented:
Changes done. Is this enough?
Which conclusions are you referring to?
The commit message character limits are clear ( although i'm not aware of the reason )
If you're referring to the clearing up of the pr body, I believe a Initial header alluding to these point would be enough. Something along the lines

```

Please don't fill below, just use it as a template or guideline to write your pr request. 
Our CI includes the PR body in the merge, so keep only what you'd like to read in a commit message.






```

( The spacing is intentional, as to have only the header message displayed in the initial viewport )

saem commented on compiler/front/main.nim:
```diff
@@ -427,10 +424,3 @@ proc mainCommand*(graph: ModuleGraph) =
       echo conf.dump(conf.vmProfileData)
     genSuccessX(conf)
 
-  when PrintRopeCacheStats:
```
I did a quick search and it seems there are some global variables left around `gCacheTries` for one.

saem reviewed (changes requested):
Found some more dead code.

Might be possible to get the stats working again if it's a matter of the global not being initialized.

What's the trace?

saem commented:
Thanks for the template review!

canelhasmateus commented:
Adding a import utils/rope fixes the problem.

trace below ( but its just standard 'not declared' shenanigans. )
Will change the pr accordingly.

```
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\front\main.nim(433, 24) Error: undeclared identifier: 'gCacheTries'
candidates (edit distance, scope distance); see '--spellSuggest':
 (6, 1): 'cache' [let declared in C:\Users\Mateus\Desktop\workspace\nimskull\compiler\front\main.nim(254, 7)]
 (6, 3): 'pathutils' [module declared in C:\Users\Mateus\Desktop\workspace\nimskull\compiler\front\main.nim(46, 8)]
 (6, 5): 'getCacheDir' [proc declared in C:\Users\Mateus\Desktop\workspace\nimskull\lib\pure\os.nim(920, 6)]
 (6, 5): 'getCacheDir' [proc declared in C:\Users\Mateus\Desktop\workspace\nimskull\lib\pure\os.nim(944, 6)]
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\nim.nim(145)               nim
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\nim.nim(86)                handleCmdLine    
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\front\main.nim(318)        mainCommand      
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\front\main.nim(288)        compileToBackend 
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\front\main.nim(139)        commandCompileToCC:\Users\Mateus\Desktop\workspace\nimskull\compiler\modules\modules.nim(213)   compileProject
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\modules\modules.nim(127)   compileModule    
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\sem\passes.nim(220)        processModule    
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\sem\passes.nim(95)         processTopLevelStmt
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\sem\sem.nim(779)           myProcess        
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\sem\sem.nim(706)           semStmtAndGenerateGenerics
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\sem\semstmts.nim(2514)     semStmt
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\sem\semexprs.nim(1160)     semExprNoType
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\sem\semexprs.nim(3230)     semExpr
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\modules\importer.nim(377)  evalImport       
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\modules\importer.nim(349)  impMod
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\modules\importer.nim(321)  myImportModule   
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\modules\modules.nim(157)   importModule     
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\modules\modules.nim(127)   compileModule    
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\sem\passes.nim(223)        processModule
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\sem\passes.nim(95)         processTopLevelStmt
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\sem\sem.nim(779)           myProcess        
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\sem\sem.nim(706)           semStmtAndGenerateGenerics
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\sem\semstmts.nim(2514)     semStmt
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\sem\semexprs.nim(1160)     semExprNoType    
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\sem\semexprs.nim(3214)     semExpr
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\sem\semstmts.nim(2233)     semProc
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\sem\semstmts.nim(2152)     semProcAux
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\sem\sempass2.nim(1516)     trackProc        
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\ast\errorreporting.nim(39) track
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\front\msgs.nim(504)        handleReport     
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\front\msgs.nim(487)        handleReport     
C:\Users\Mateus\Desktop\workspace\nimskull\compiler\front\msgs.nim(368)        quit
The terminal process "C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -Command ./bin/nim.exe --debugger:native --out:C:\Users\Mateus\Desktop\workspace\nimskull\bin\mynim.exe c ./compiler/nim.nim" terminated with exit code: 1.
```

saem reviewed (approved):
bors r+

saem commented:
bors cancel

canelhasmateus committed:
Fixed PrintRopeStats.

PrintRopeCacheStats is declared as a const false,
that acted as a guardian to a when block that references undeclared variables.
As such, setting it to true crashes the compiler.
Importing ropes.nim fixes the issue.

saem commented on compiler/front/main.nim:
```diff
@@ -223,6 +223,9 @@ proc commandView(graph: ModuleGraph) =
   PrintRopeCacheStats = false
 
+when PrintRopeCacheStats:
+    import utils/ropes
```
```suggestion
  import utils/ropes
```
fixing the indentation, this should be compiler code base standardizes on two space indent.

saem commented:
bors r+
